File: python/ABSTRACT_FACTORY_DECORATOR/ABSTRACT_FACTORY_DECORATOR/reservacion/ReservacionMultiple.py
from .Reservacion import Reservacion

class ReservacionMultiple(Reservacion):
    
    def __init__(self):
        self.reservaciones = []
    
    def agregarReservacion(self, reservacion):
        self.reservaciones.append(reservacion)
    
    def eliminarReservacion(self, reservacion):
        self.reservaciones.remove(reservacion)
        
    def getReservaciones(self):
        return self.reservaciones
    
    def mostrarReservacion(self):
        for i, reservacion in enumerate(self.reservaciones):
            print(f'Reservacion[{i}]\n')
            reservacion.mostrarReservacion()

File: php/ABSTRACT_FACTORY_DECORATOR/reservacion/ReservacionMultiple.php
<?php
require_once 'Reservacion.php';
class ReservacionMultiple extends Reservacion{
    public $reservaciones;
    
    public function __construct() {
        $this->reservaciones = array();
    }
    
    public function agregarReservacion($reservacion) {
        array_push($this->reservaciones, $reservacion);
    }
    
    public function eliminarReservacion($reservacion) {
        unset($this->reservaciones[array_search($reservacion, $this->reservaciones)]);
    }
    public function MostrarReservaciones() {
        foreach ($this->reservaciones as $r) {
            
            echo "<ul><li>".$r->mostrarReservacion()."</li></ul>";
        }
    }
    public function getReservaciones() {
        return $this->reservaciones;
    }
}
